ihatemoney/utils.py

```diff
@@ -131,3 +132,38 @@ def list_of_dicts2csv(dict_to_convert):
 
 # base64 encoding that works with both py2 and py3 and yield no warning
 base64_encode = base64.encodestring if six.PY2 else base64.encodebytes
+
+
+class LoginThrottler():
+    """Simple login throttler used to limit authentification attempts based on client's ip address
+    When using multiple workers, remaining number of attempts can get inconsistent
+    but will still be limited to num_workers * max_attempts
+    """
+    def __init__(self, max_attempts=3, delay=1):
+        self._max_attempts = max_attempts
+        # Delay in minutes before attempts counter is reset
+        self._delay = delay
+        self._attempts = {}
+
+    def get_remaining_attempts(self, ip):
+        return self._max_attempts - self._attempts.get(ip, [datetime.now(), 0])[1]
+
+    def increment_attempts_counter(self, ip):
+        if self._attempts.get(ip) is None:
+            # Store first attempt date and number of attempts since
+            self._attempts[ip] = [datetime.now(), 0]
```

Comment:
I'm a bit concerned with this behavior. I might be too worried for nothing, but this makes it possible to overwhelm the memory of the server by faking a lot of requests from different IP addresses. In order to fix this, we could free the memory (reset the dict) once we hit a big number of IP addresses (let's say 10k IPs ?). What do you think?